iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java

```diff
@@ -44,11 +48,28 @@ void isEqualTo() {
     assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(notTextTree())).isFalse();
   }
 
+  @Test
+  void isSet() {
+    assertThat(TreePredicates.isSet().test(yaml(List.of("")))).isFalse();
+    assertThat(TreePredicates.isSet().test(yaml(List.of("", "~")))).isFalse();
+    assertThat(TreePredicates.isSet().test(yaml(List.of("a")))).isTrue();
+    assertThat(TreePredicates.isSet().test(yaml(List.of("a", "")))).isTrue();
+    assertThat(TreePredicates.isSet().test(notTextTree())).isFalse();
+  }
+
   private ScalarTree text(String value) {
     return new ScalarTreeImpl(value, null, null);
   }
 
   private FileTreeImpl notTextTree() {
     return new FileTreeImpl(null, null);
   }
+
+  private YamlTree yaml(List<String> values) {
+    if (values.size() <= 1) {
+      return new ScalarTreeImpl(values.stream().findFirst().orElse(""), null, null);
+    }
+    List<YamlTree> elements = values.stream().map(v -> new ScalarTreeImpl(v, null, null)).collect(Collectors.toList());
+    return new SequenceTreeImpl(elements, null);
+  }
```

Comment:
Same comment as above. Since both test classes are in the same module, you could also move this method to a utils class.